diff --git a/gfx/gfx_display.c b/gfx/gfx_display.c
index 897326c321..3791245576 100644
--- a/gfx/gfx_display.c
+++ b/gfx/gfx_display.c
@@ -29,7 +29,7 @@
 
 /* 'OZONE_SIDEBAR_WIDTH' must be kept in sync
  * with Ozone driver metrics */
-#define OZONE_SIDEBAR_WIDTH 408
+#define OZONE_SIDEBAR_WIDTH 240
 
 /* Small 1x1 white texture used for blending purposes */
 static uintptr_t gfx_white_texture;
@@ -316,17 +316,6 @@ float gfx_display_get_dpi_scale(
        || (p_disp->menu_driver_id != last_menu_driver_id))
    {
       adjusted_scale            = scale * menu_scale_factor;
-#ifdef HAVE_OZONE
-      if (p_disp->menu_driver_id == MENU_DRIVER_ID_OZONE)
-      {
-         /* Ozone has a capped scale factor */
-         float new_width        = (float)width * 0.3333333f;
-         if (((float)OZONE_SIDEBAR_WIDTH * adjusted_scale)
-               > new_width)
-            adjusted_scale      = (new_width / (float)OZONE_SIDEBAR_WIDTH);
-      }
-#endif
-      adjusted_scale            = (adjusted_scale > 0.0001f) ? adjusted_scale : 1.0f;
       last_menu_scale_factor    = menu_scale_factor;
       last_menu_driver_id       = p_disp->menu_driver_id;
    }
diff --git a/menu/drivers/ozone.c b/menu/drivers/ozone.c
index 1cdf1b18cd..4104cac8b1 100644
--- a/menu/drivers/ozone.c
+++ b/menu/drivers/ozone.c
@@ -80,8 +80,8 @@
 #define HEADER_HEIGHT                 87
 #define FOOTER_HEIGHT                 78
 
-#define ENTRY_PADDING_HORIZONTAL_HALF 40
-#define ENTRY_PADDING_HORIZONTAL_FULL 140
+#define ENTRY_PADDING_HORIZONTAL_HALF 10
+#define ENTRY_PADDING_HORIZONTAL_FULL 10
 #define ENTRY_PADDING_VERTICAL        20
 #define ENTRY_HEIGHT                  50
 #define ENTRY_SPACING                 8
@@ -90,8 +90,8 @@
 
 /* > 'SIDEBAR_WIDTH' must be kept in sync with
  *   menu driver metrics */
-#define SIDEBAR_WIDTH                 408
-#define SIDEBAR_X_PADDING             40
+#define SIDEBAR_WIDTH                 240
+#define SIDEBAR_X_PADDING             10
 #define SIDEBAR_Y_PADDING             20
 #define SIDEBAR_ENTRY_HEIGHT          50
 #define SIDEBAR_ENTRY_Y_PADDING       10
